Source/Scene/OrderedGroundPrimitiveCollection.js

```diff
@@ -0,0 +1,230 @@
+define([
+    '../Core/defaultValue',
+    '../Core/defined',
+    '../Core/defineProperties',
+    '../Core/destroyObject',
+    '../Core/Check',
+    '../Core/DeveloperError',
+    './PrimitiveCollection'
+], function(
+    defaultValue,
+    defined,
+    defineProperties,
+    destroyObject,
+    Check,
+    DeveloperError,
+    PrimitiveCollection) {
+    'use strict';
+
+    /**
+     * A primitive collection for helping maintain the order or ground primitives based on a z-index
+     *
+     * @private
+     */
+    function OrderedGroundPrimitiveCollection() {
+        this._length = 0;
```

Comment:
I think you can shorten/simplify this class by using `AssociativeArray`, you've basically rolled your own here.

Reply:
Nope, `AssociativeArray` has no concept of order and that's critical for this to work

Reply:
Ahh, you're absolutely correct.  Sorry.